crates/bevy_ecs/src/lib.rs

```diff
@@ -660,78 +678,86 @@ mod tests {
+        {
             if i % 2 == 0 {
                 a.0 += 1;
             }
         }
 
-        fn get_filtered<F: WorldQuery>(world: &mut World) -> Vec<Entity>
+        fn get_filtered<F: WorldQuery>(world: &mut World, system_counter: u32) -> Vec<Entity>
         where
             F::Fetch: FilterFetch,
         {
             world
                 .query_filtered::<Entity, F>()
-                .iter(&world)
+                .iter(&world, Some(system_counter))
                 .collect::<Vec<Entity>>()
         }
 
-        assert_eq!(get_filtered::<Mutated<A>>(&mut world), vec![e1, e3]);
+        assert_eq!(get_filtered::<Mutated<A>>(&mut world, 0), vec![e1, e3]);
 
         // ensure changing an entity's archetypes also moves its mutated state
         world.entity_mut(e1).insert(C);
 
-        assert_eq!(get_filtered::<Mutated<A>>(&mut world), vec![e3, e1], "changed entities list should not change (although the order will due to archetype moves)");
+        assert_eq!(get_filtered::<Mutated<A>>(&mut world, 0), vec![e3, e1], "changed entities list should not change (although the order will due to archetype moves)");
 
         // spawning a new A entity should not change existing mutated state
         world.entity_mut(e1).insert_bundle((A(0), B));
         assert_eq!(
-            get_filtered::<Mutated<A>>(&mut world),
+            get_filtered::<Mutated<A>>(&mut world, 0),
             vec![e3, e1],
             "changed entities list should not change"
         );
 
         // removing an unchanged entity should not change mutated state
         assert!(world.despawn(e2));
         assert_eq!(
-            get_filtered::<Mutated<A>>(&mut world),
+            get_filtered::<Mutated<A>>(&mut world, 0),
             vec![e3, e1],
             "changed entities list should not change"
         );
 
         // removing a changed entity should remove it from enumeration
         assert!(world.despawn(e1));
         assert_eq!(
-            get_filtered::<Mutated<A>>(&mut world),
+            get_filtered::<Mutated<A>>(&mut world, 0),
             vec![e3],
             "e1 should no longer be returned"
         );
 
         world.clear_trackers();
+        world.increment_global_system_counter();
 
-        assert!(get_filtered::<Mutated<A>>(&mut world).is_empty());
+        assert!(get_filtered::<Mutated<A>>(&mut world, 1).is_empty());
 
         let e4 = world.spawn().id();
 
         world.entity_mut(e4).insert(A(0));
-        assert!(get_filtered::<Mutated<A>>(&mut world).is_empty());
-        assert_eq!(get_filtered::<Added<A>>(&mut world), vec![e4]);
+        assert!(get_filtered::<Mutated<A>>(&mut world, 1).is_empty());
+        assert_eq!(get_filtered::<Added<A>>(&mut world, 1), vec![e4]);
 
         world.entity_mut(e4).insert(A(1));
-        assert_eq!(get_filtered::<Mutated<A>>(&mut world), vec![e4]);
+        // assert_eq!(get_filtered::<Mutated<A>>(&mut world, 1), vec![e4]); // This case is no longer possible to detect
```

Comment:
I'd like to bring attention to this point. `Mutated` is now defined as "`Changed` but not `Added`", so it cannot detect if something was both added and mutated in the same (exclusive) system execution.
I'm thinking about removing `Mutated`, cart seemed to agree that it's not very useful. I'll add the `Not` query transformer at the same time, such that `Query<T, (Changed<T>, Not<Added<T>>)>` is still possible if you do need it.

Reply:
I strongly agree with both of those proposals. `Not` in particular seems like a very natural extension.